downstream/dsurf/_Surface.py

```diff
@@ -1,25 +1,37 @@
+import types
 import typing
 
-from ._Policy import Policy
-
 
 class Surface:
 
-    _storage: list  # storage sites
+    __slots__ = ("_storage", "algo", "T")
+
+    algo: types.ModuleType
+    _storage: typing.MutableSequence  # storage sites
     T: int  # current logical time
-    policy: Policy  # policy
 
-    def __init__(self: "Surface", policy: Policy) -> None:
+    def __init__(
+        self: "Surface",
+        algo: types.ModuleType,
+        storage: typing.Union[typing.MutableSequence[object], int],
+    ) -> None:
```

Comment:
We should add a brief docstring (especially explaining the int argument)

Reply:
(and probably explicitly suggesting that a numpy array can be used for efficient storage)